File: app/oidc/config/LineOidcConfiguration.java
package oidc.config;

import org.pac4j.oidc.config.OidcConfiguration;
import com.nimbusds.jose.JWSAlgorithm;

// Directly add jwsAlgorithm, 
// Because LINE only support HS256 jwsAlgorithm for id token, and pac4j configuration do not read id token header's alg 
public class LineOidcConfiguration extends OidcConfiguration {
    public void setIDTokenJwsAlgorithm(final JWSAlgorithm jwsAlgorithm) {
        var meta = this.findProviderMetadata();
        meta.getIDTokenJWSAlgs().add(jwsAlgorithm);
        this.setProviderMetadata(meta);
    }

    public void setIDTokenJwsAlgorithm(final String jwsAlgorithm) {
        var meta = this.findProviderMetadata();
        meta.getIDTokenJWSAlgs().add(JWSAlgorithm.parse(jwsAlgorithm));
        this.setProviderMetadata(meta);
    }
}

File: app/oidc/config/LineOidcConfiguration.scala
package oidc.config

import com.nimbusds.jose.JWSAlgorithm
import org.pac4j.oidc.config.OidcConfiguration

// Directly add jwsAlgorithm,
// Because LINE only support HS256 jwsAlgorithm for id token, and pac4j configuration do not read id token header's alg
class LineOidcConfiguration extends OidcConfiguration {

  def setIDTokenJwsAlgorithm(jwsAlgorithm: JWSAlgorithm): Unit = {
    val meta = this.findProviderMetadata()
    meta.getIDTokenJWSAlgs().add(jwsAlgorithm)
    this.setProviderMetadata(meta)
  }

  def setIDTokenJwsAlgorithm(jwsAlgorithmString: String): Unit = {
    val meta = this.findProviderMetadata()
    meta.getIDTokenJWSAlgs().add(JWSAlgorithm.parse(jwsAlgorithmString))
    this.setProviderMetadata(meta)
  }
}
